### PR TITLE
fix bug in wasi::environ_get, fix off by one error in env_size_get

### LINKED ISSUE #475: Ability to set environment variables
I couldn't find a way to set an environment variable neither in the help nor in the Rust code.

### Motivation

Because it's an easy way to configure a program's behavior.

### Proposed solution

Add a `--env key=value` cli option.

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
adds `--env` flag (resolves #475)
fixes other wasi env bugs
adds wasi envvar test

### CODE REVIEW

**Comment by syrusakbary** [comment]
That's a great finding.

I think it will be good to:
* Add integration tests
* Only allow part of the environment variables to leak via a new CLI option

File: lib/wasi/wasitests/envvar.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    let mut env_vars = env::vars()
+        .map(|(key, value)| format!("{}={}", key, value))
+        .collect::<Vec<String>>();
+
+    env_vars.sort();
+
+    println!("Env vars:");
+    for e in env_vars {
+        println!("{}", e);
+    }
```
Comment (syrusakbary) [inline lib/wasi/wasitests/envvar.rs]: Can we add a case to retrieve a single env var when:
* env var is present (such as `DOG`)
* env var is not present (such as `UNKNOWN`)

**Comment by MarkMcCaskey** [comment]
bors try

**Pull request merged**